### PR TITLE
GAP-1975: Update 'Sign up for grant updates' Newsletter to use One Login

### PR DESCRIPTION
**Pull request by ConorFayleTCO**
## Description

[GAP-1975](https://technologyprogramme.atlassian.net/browse/GAP-1975)

Get subscribe and unsubscribe journeys working for One Login, update readme and .gitignore

https://github.com/cabinetoffice/gap-find-backend/pull/65

## Type of change

Please check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes:

- [ ] Unit Test

- [ ] Integration Test (if applicable)

- [ ] End to End Test (if applicable)

## Screenshots (if appropriate):

Please attach screenshots of the change if it is a UI change:

# Checklist:

- [ ] If I have listed dependencies above, I have ensured that they are present in the target branch.
- [x] I have performed a self-review of my code.
- [x] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation where applicable.
- [ ] I have ran cypress tests and they all pass locally.

[GAP-1975]: https://technologyprogramme.atlassian.net/browse/GAP-1975?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pages/unsubscribe/[id].tsx [hunk 1]
```diff
@@ -115,12 +123,14 @@ const handleUnsubscribe = async (
   notificationKey: NotificationKey,
   emailAddress: string,
   unsubscribeReferenceId: string,
-) =>
-  UNSUBSCRIBE_HANDLER_MAP[type](
+) => {
+  console.log({ type, notificationKey, emailAddress, unsubscribeReferenceId });
```
Comment (ConorFayleTCO) [inline pages/unsubscribe/[id].tsx:127]:
```suggestion
```

File: pages/unsubscribe/[id].tsx [hunk 2]
```diff
@@ -32,6 +32,13 @@ export async function getServerSideProps({ query: { id = '' } = {} }) {
     emailAddress = await decrypt(encryptedEmailAddress);
     notificationId = subscriptionId ?? newsletterId ?? savedSearchId;
 
+    console.log({
+      encryptedEmailAddress,
+      emailAddress,
+      notificationType,
+      notificationId,
+    });
+
```
Comment (ConorFayleTCO) [inline pages/unsubscribe/[id].tsx:41]:
```suggestion
```

File: src/service/newsletter/newsletter-subscription-service.ts [hunk 3]
```diff
@@ -23,22 +30,55 @@ export class NewsletterSubscriptionService {
...
+  }
+
   async unsubscribeFromNewsletter(
     plaintextEmail: string,
     type: NewsletterType,
     unsubscribeReferenceId?: string,
+    sub?: string,
   ): Promise<void> {
+    const id = sub ?? plaintextEmail;
+    console.log(id);
```
Comment (ConorFayleTCO) [inline src/service/newsletter/newsletter-subscription-service.ts:76]:
```suggestion
```

File: pages/unsubscribe/[id].tsx [hunk 4]
```diff
@@ -91,6 +91,7 @@ const newsletterHandler = async (
 ) => {
   const newsletterSubscriptionService =
     NewsletterSubscriptionService.getInstance();
+  console.log('handler', id, emailAddress, unsubscribeReferenceId);
```
Comment (ConorFayleTCO) [inline pages/unsubscribe/[id].tsx:94]:
```suggestion
```

File: pages/newsletter/index.tsx [hunk 5]
```diff
@@ -1,20 +1,31 @@
...
 
+const HOST = process.env.HOST;
+const USER_SERVICE_HOST = process.env.USER_SERVICE_HOST;
 export async function getServerSideProps({ query }) {
-  return { props: { returnParams: query } };
+  const ONE_LOGIN_ENABLED = process.env.ONE_LOGIN_ENABLED === 'true';
+  return {
+    props: {
+      returnParams: query,
+      ONE_LOGIN_ENABLED,
```
Comment (jgunnCO) [inline pages/newsletter/index.tsx:16]: keep same naming convention as other env config?

```suggestion
      oneLoginEnabled: ONE_LOGIN_ENABLED,
```

File: src/components/confirmation-message/ConfirmationMessage.js [hunk 6]
```diff
@@ -20,8 +20,9 @@ function ConfirmationMessage({ message }) {
             className="govuk-notification-banner__heading"
             data-cy="cySubscribeSuccessMessageContent"
           >
-            {message}
+            {heading}
           </h3>
+          {subheading ? <p className="gov-body">{subheading}</p> : null}
```
Comment (jgunnCO) [inline src/components/confirmation-message/ConfirmationMessage.js:25]:
```suggestion
          {subheading && <p className="gov-body">{subheading}</p>}
```

**Review by jgunnCO — APPROVED** [review]

**Pull request merged**